mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -884,9 +890,20 @@ LogicalResult ModuleImport::convertGlobal(llvm::GlobalVariable *globalVar) {
     globalExpressionAttr =
         debugImporter->translateGlobalVariableExpression(globalExpressions[0]);
 
+  // Workaround to support LLVM's nameless globals. MLIR, in contrast to LLVM,
+  // always requires a symbol name.
+  std::string globalName = globalVar->getName().str();
+  if (globalName.empty()) {
+    // Make sure the symbol name does not clash with an existing symbol.
+    do {
+      globalName =
+          getNamelessGlobalPrefix().str() + std::to_string(namelessGlobalId++);
+    } while (llvmModule->getNamedValue(globalName));
```

Comment:
Works for me. The SymbolTable also has a static function for this `generateSymbolName`. I think I commented before but I don't see the comment now... So I am trying to repost in case it got lost in GitHub somewhere.

The amount of code saved is not too large so feel free to keep this if the exact format of the name is relevant.